File: eCinema/eCinema.UI/ecinema_mobile/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int id;
  final int bookingId;
  final int amount;
  final String currency;
  final String stripePaymentIntentId;
  final String? stripeChargeId;

  @JsonKey(fromJson: paymentStatusFromString)
  final PaymentStatus status;

  final DateTime createdAt;
  final DateTime? succeededAt;
  final DateTime? failedAt;
  final DateTime? refundedAt;

  Payment({
    required this.id,
    required this.bookingId,
    required this.amount,
    required this.currency,
    required this.stripePaymentIntentId,
    this.stripeChargeId,
    required this.status,
    required this.createdAt,
    this.succeededAt,
    this.failedAt,
    this.refundedAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}

enum PaymentStatus { pending, succeeded, failed, refunded, unknown }

PaymentStatus paymentStatusFromString(dynamic statusValue) {
  if (statusValue == null) return PaymentStatus.unknown;

  if (statusValue is String) {
    switch (statusValue.toLowerCase()) {
      case 'pending':
        return PaymentStatus.pending;
      case 'succeeded':
        return PaymentStatus.succeeded;
      case 'failed':
        return PaymentStatus.failed;
      case 'refunded':
        return PaymentStatus.refunded;
      default:
        return PaymentStatus.unknown;
    }
  } else if (statusValue is int) {
    switch (statusValue) {
      case 0:
        return PaymentStatus.pending;
      case 1:
        return PaymentStatus.succeeded;
      case 2:
        return PaymentStatus.failed;
      case 3:
        return PaymentStatus.refunded;
      default:
        return PaymentStatus.unknown;
    }
  }
  return PaymentStatus.unknown;
}

File: eCinema/eCinema.Models/Entities/Payment.cs
using eCinema.Models.Entities;
using System;
using System.Text.Json.Serialization;

namespace CinemaApp.Domain.Entities
{
    public class Payment
    {
        public int Id { get; set; }
        public int BookingId { get; set; }
        public Booking Booking { get; set; }

        public long Amount { get; set; }
        public string Currency { get; set; } = "bam";

        public string StripePaymentIntentId { get; set; }
        public string? StripeChargeId { get; set; }

        public PaymentStatus Status { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? SucceededAt { get; set; }
        public DateTime? FailedAt { get; set; }
        public DateTime? RefundedAt { get; set; }
    }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum PaymentStatus
    {
        Pending = 0,
        Succeeded = 1,
        Failed = 2,
        Refunded = 3
    }

}
